File: abc/abc320/a.py
# https://atcoder.jp/contests/abc320/submissions/45584000

a, b = map(int, input().split())
ans = 1
base = 1
for i in range(b):
  base *= a
ans = base
base = 1
for i in range(a):
  base *= b
ans += base
print(ans)

File: abc/abc320/a.java
// https://atcoder.jp/contests/abc320/submissions/45741497

import java.util.Scanner;

class Main {
 static int pow(int a, int b) {
    int res = 1;
    for(int i = 0; i < b; i++) {
      res *= a;
    }
    return res;
  }

  public static void main(String args[]) {
    Scanner stdIn = new Scanner(System.in);

    int a = stdIn.nextInt();
    int b = stdIn.nextInt();

    System.out.println(pow(a, b) + pow(b, a));
  }
}

